src/forms/datepicker/DatePicker.tsx

```diff
@@ -0,0 +1,423 @@
+                                                    {navValue.format("YYYY")}
+                                                </button>
+                                            </div>
+                                            <div className="calendar-switchers">
+                                                <button
+                                                    className="calendar-btn switcher"
+                                                    type="button"
+                                                    onClick={this.toggleNavValueMonth.bind(this, navValue.month() - 1)}
+                                                >
+                                                    <Icon shape="angle" dir={Direction.LEFT} />
+                                                </button>
+                                                <button
+                                                    className="calendar-btn switcher"
+                                                    type="button"
+                                                    onClick={this.toggleNavValue.bind(this)}
+                                                >
+                                                    <Icon shape="event" />
+                                                </button>
+                                                <button
+                                                    className="calendar-btn switcher"
+                                                    type="button"
+                                                    onClick={this.toggleNavValueMonth.bind(this, navValue.month() + 1)}
+                                                >
+                                                    <Icon shape="angle" dir={Direction.RIGHT} />
+                                                </button>
+                                            </div>
+                                        </div>
+                                        <table className="calendar-table weekdays">
+                                            <tbody>
+                                                <tr className="calendar-row">
+                                                    {Moment.weekdaysShort(true).map((day: string) => {
+                                                        return (
+                                                            <td key={"day_" + day} className="calendar-cell weekday">
+                                                                {day.substring(0, 1)}
+                                                            </td>
+                                                        );
+                                                    })}
+                                                </tr>
+                                            </tbody>
+                                        </table>
+                                        <table className="calendar-table calendar-dates">
```

Comment:
Table adds nesting. Can we move it to other components to make render more readable and encapsulated?

Reply:
I did some refactoring here.  What do you think now?